source/fundamentals/data-formats/custom-types.txt

```diff
@@ -1,233 +1,230 @@
-.. uses custom-type.rst
+.. _pymongo-custom-types:
 
-Custom Type Example
-===================
+============
+Custom Types
+============
 
-This is an example of using a custom type with PyMongo. The example here shows
-how to subclass ``~bson.codec_options.TypeCodec`` to write a type
-codec, which is used to populate a ``~bson.codec_options.TypeRegistry``.
-The type registry can then be used to create a custom-type-aware
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: tutorial
+
+.. meta::
+   :keywords: codec, serialize, deserialize
+
+Overview
+--------
+
+In this guide, you can learn how to create a custom type with {+driver-short+}.
+This guide describes how to subclass ``~bson.codec_options.TypeCodec`` to write a type
+codec, which you can use to populate a ``~bson.codec_options.TypeRegistry``.
+You can then use the type registry to create a custom-type-aware
 ``~pymongo.collection.Collection``. Read and write operations
 issued against the resulting collection object transparently manipulate
 documents as they are saved to or retrieved from MongoDB.
 
-
 Setting Up
 ----------
 
-We'll start by getting a clean database to use for the example:
+First, create a new database:
 
 .. code-block:: python
 
-  >>> from pymongo import MongoClient
-  >>> client = MongoClient()
-  >>> client.drop_database("custom_type_example")
-  >>> db = client.custom_type_example
+   >>> from pymongo import MongoClient
+   >>> client = MongoClient()
+   >>> client.drop_database("custom_type_example")
+   >>> db = client.custom_type_example
 
 
-Since the purpose of the example is to demonstrate working with custom types,
-we'll need a custom data type to use. For this example, we will be working with
-the ``~decimal.Decimal`` type from Python's standard library. Since the
-BSON library's ``~bson.decimal128.Decimal128`` type (that implements
-the IEEE 754 decimal128 decimal-based floating-point numbering format) is
+This example uses the ``~decimal.Decimal`` type from Python's standard library. Since the
+BSON library's ``~bson.decimal128.Decimal128`` type--that implements
+the ``IEEE 754 decimal128 decimal-based floating-point`` numbering format--is
 distinct from Python's built-in ``~decimal.Decimal`` type, attempting
-to save an instance of ``Decimal`` with PyMongo, results in an
+to save an instance of ``Decimal`` with PyMongo, results in a
```

Comment:
Can't make a suggestion on it in GH, but I would change the end of line 48 to" "type. Trying"